src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -1,214 +1,144 @@
-    refreshPanelsVisibility()
+    allContentPanel.add(signInWithSourcegraphPanel, SIGN_IN_PANEL, SIGN_IN_PANEL_INDEX)
+    allContentLayout.show(allContentPanel, SIGN_IN_PANEL)
 
+    ApplicationManager.getApplication().invokeLater { refreshPanelsVisibility() }
     ApplicationManager.getApplication().executeOnPooledThread { refreshSubscriptionTab() }
-    refreshChatToEmpty()
-    historyTree.refreshTree()
+
+    startNewChat(project)
   }
 
   @RequiresBackgroundThread
   fun refreshSubscriptionTab() {
     CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
-      fetchSubscriptionPanelData(project, agent.server).thenAccept {
-        if (it != null) {
-
-          ApplicationManager.getApplication().invokeLater { refreshSubscriptionTab(it) }
+      val data = fetchSubscriptionPanelData(project, agent.server)
+      if (data != null) {
+        val isSubscriptionTabPresent = tabbedPane.tabCount >= SUBSCRIPTION_TAB_INDEX + 1
+        if (data.isDotcomAccount && data.codyProFeatureFlag) {
+          if (!isSubscriptionTabPresent) {
+            tabbedPane.insertTab(
+                /* title = */ "Subscription",
+                /* icon = */ null,
+                /* component = */ subscriptionPanel,
+                /* tip = */ null,
+                SUBSCRIPTION_TAB_INDEX)
+          }
+          subscriptionPanel.update(data.isCurrentUserPro)
+        } else if (isSubscriptionTabPresent) {
+          tabbedPane.remove(SUBSCRIPTION_TAB_INDEX)
         }
       }
     }
   }
 
-  @RequiresEdt
-  private fun refreshSubscriptionTab(data: SubscriptionTabPanelData) {
-    val isSubscriptionTabPresent = tabbedPane.tabCount >= SUBSCRIPTION_TAB_INDEX + 1
-
-    if (data.isDotcomAccount && data.codyProFeatureFlag) {
-      if (!isSubscriptionTabPresent) {
-        tabbedPane.insertTab(
-            /* title = */ "Subscription",
-            /* icon = */ null,
-            /* component = */ subscriptionPanel,
-            /* tip = */ null,
-            SUBSCRIPTION_TAB_INDEX)
+  private fun runNewCommand(project: Project, commandId: CommandId) {
+    startNewPanel(project) { agent: CodyAgent ->
+      when (commandId) {
+        CommandId.Explain -> agent.server.commandsExplain()
+        CommandId.Smell -> agent.server.commandsSmell()
+        CommandId.Test -> agent.server.commandsTest()
       }
-      subscriptionPanel.update(data.isCurrentUserPro)
-    } else if (isSubscriptionTabPresent) {
-      tabbedPane.remove(SUBSCRIPTION_TAB_INDEX)
     }
   }
 
-  fun refreshChatToEmpty() {
-    CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
-      ApplicationManager.getApplication().invokeLater {
-        messagesPanel.removeAll()
-        chatMessageHistory.clearHistory()
-        inProgressChat.abort()
-        addWelcomeMessage()
-        activateChatTab()
-      }
-      chatId = agent.server.chatNew().get()
-    }
+  private fun startNewChat(project: Project) {
```

Comment:
`project` is a private val of this class already - there is no need to pass it here